pkg/controller/controller_test.go

```diff
@@ -2781,3 +2747,190 @@ func TestProvisionFromPVC(t *testing.T) {
 		}
 	}
 }
+
+func TestProvisionWithMigration(t *testing.T) {
+	var requestBytes int64 = 100000
+	var inTreePluginName = "in-tree-plugin"
+
+	deletePolicy := v1.PersistentVolumeReclaimDelete
+	testcases := []struct {
+		name              string
+		provisioner       string
+		annotation        map[string]string
+		expectTranslation bool
+		expectCreate      bool
+		expectErr         bool
+	}{
+		{
+			name:              "provision with migration on",
+			provisioner:       inTreePluginName,
+			annotation:        map[string]string{annStorageProvisioner: driverName},
+			expectCreate:      true,
+			expectTranslation: true,
+		},
+		{
+			name:              "provision without migration for native CSI",
+			provisioner:       driverName,
+			annotation:        map[string]string{annStorageProvisioner: driverName},
+			expectCreate:      true,
+			expectTranslation: false,
+		},
+		{
+			name:        "ignore in-tree PVC when provisioned by in-tree",
+			provisioner: inTreePluginName,
+			annotation:  map[string]string{annStorageProvisioner: inTreePluginName},
+			expectErr:   true,
+		},
+	}
+
+	for k, tc := range testcases {
+		t.Logf("running test: %s", tc.name)
```

Comment:
It's preferred to use `t.Run` for table-driven tests. Then you can completely ignore the `k` value that you log in certain cases below. Any test log or test error will have the context of the subtest name.